### DISCUSSION TITLE
SymbolLayer  iconImage

### DISCUSSION THREAD

**Discussion by Kovalskiii** [rnmapbox/maps#2916] [Q&A] [answered]
Hello. I implemented clustered markers. But i am confused how to implement change marker iconImage (in SymbolLayer) by eventType from features properties. So let's imagine, i have several types and images for this types, and i want to set images on marker according this types.  This is my code: 
[markers.txt](https://github.com/rnmapbox/maps/files/11790311/markers.txt)

-
`  return (
    <View style={styles.page}>
      <View style={styles.container}>
        <Mapbox.MapView
          styleURL={"mapbox://styles/mapbox/dark-v11"}
          style={styles.map}
          rotateEnabled={false}
        >
          {eventMarkersData && (
            <Mapbox.ShapeSource
              ref={shapeSource}
              id="events"
              cluster={true}
              clusterMaxZoom={18}
              clusterRadius={60}
              shape={eventMarkersData}
              onPress={async marker => {
                if (marker.features[0].properties.cluster) {
                  const collection = await shapeSource.current.getClusterLeaves(
                    marker.features[0],
                    marker.features[0].properties.point_count,
                    0,
                  )
                  // Do what you want if the user clicks the cluster
                  console.log('Cluster',collection)
                } else {
                  // Do what you want if the user clicks individual marker
                  console.log('One marker', marker)
                  console.log('One marker', marker.features[0].properties)
                }
              }}
            >

              <Mapbox.Images>
                <Mapbox.Image name="clusterImg">
                  <View style={styles.markerImageContainer} pointerEvents="none">
                    <NightClub width={24} height={24} fill='#E4E4E4'/>
                  </View>
                </Mapbox.Image>
                <Mapbox.Image name="singleMarkerImage">
                  <View style={styles.markerImageContainer} pointerEvents="none">
                    <Yacht width={24} height={24} fill='#E4E4E4'/>
                  </View>
                </Mapbox.Image>
              </Mapbox.Images>

              {/*Clustered marker image*/}
              <Mapbox.SymbolLayer
                  id="clusteredMarkersImage"
                  filter={['has', 'point_count']}
                  style={styles.clusteredMarkersImage}
              />

              {/*Clustered marker image circle*/}
              <Mapbox.CircleLayer
                  id="clusteredMarkerImageCircle"
                  belowLayerID="clusteredMarkersImage"
                  filter={['has', 'point_count']}
                  style={styles.clusteredMarkersImageCircle}
              />

              {/*Cluster marker count*/}
              <Mapbox.SymbolLayer
                id="clusteredMarkerCount"
                style={styles.markerCount}
                filter={['has', 'point_count']}
              />

              {/*Single marker image*/}
              <Mapbox.SymbolLayer
                  id="singleMarkerImage"
                  filter={['!', ['has', 'point_count']]}
                  style={{
                    iconImage: 'singleMarkerImage',
                    iconAllowOverlap: false,
                    iconIgnorePlacement: false
                  }}
              />
            </Mapbox.ShapeSource>
          )}
        </Mapbox.MapView>
      </View>
    </View>
  );`

**Comment by Kovalskiii** [comment, marked as answer]
**So, i have already found solution** 
`   <Mapbox.Images>
                <Mapbox.Image name={eventMarkerIconHelper().iconName} sdf={true}>
                    <Airdrop width={24} height={24} fill='#E4E4E4'/>
                </Mapbox.Image>
                {eventTypes.map(type => (
                    <Mapbox.Image key={type.id} name={eventMarkerIconHelper(type.value).iconName} sdf={true}>
                        {eventMarkerIconHelper(type.value).icon}
                    </Mapbox.Image>
                ))}
              </Mapbox.Images>
`
              
`  singleMarkerIcon: {
    iconImage: ['get', 'eventTypeIconName'],
    iconOpacity: 1,
    iconColor: '#E4E4E4',
    iconAllowOverlap: true,
    iconIgnorePlacement: false,
  },`

**Comment by tomskopek** [comment]
Hey @Kovalskiii 

I'm trying to get the iconColor to work with sdf icons. Did you get that to work?
